src/utils/api/helper.ts

```diff
@@ -288,21 +288,35 @@ export const activePaymentTask = (
                 const cartInfo = getSessionItem(SessionItems.cart) as
                   | Cart
                   | undefined;
+                const paymentInfo = getSessionItem(SessionItems.paymentInfo) as
+                  | PaymentInfo
+                  | undefined;
                 if (cartInfo !== undefined) {
                   const rptIdAmountMap = new Map(
                     responseType.value.payments.map(
                       (p) => [p.rptId, p.amount] as [RptId, AmountEuroCents]
                     )
                   );
+                  const creditorReferenceMap = new Map(
+                    responseType.value.payments.map(
+                      (p) =>
+                        [p.rptId, p.creditorReferenceId] as [
+                          RptId,
+                          string | undefined
+                        ]
+                    )
+                  );
 
                   const updatedPaymentNotices = cartInfo.paymentNotices.map(
                     (paymentNotice) => {
-                      const updatedAmount = rptIdAmountMap.get(
-                        `${paymentNotice.fiscalCode}${paymentNotice.noticeNumber}` as RptId
-                      );
+                      const rptId =
+                        `${paymentNotice.fiscalCode}${paymentNotice.noticeNumber}` as RptId;
+                      const updatedAmount = rptIdAmountMap.get(rptId);
+                      const creditorId = creditorReferenceMap.get(rptId);
```

Comment:
`creditorReferenceId` is not required in the response, shouldn't `creditorId` fallback to `rptId` to be shown later?

Reply:
Yes `creditorReferenceId` is nullable, also for the whole PaymentNotice object. In this helper method I'm saving creditor reference id as nullable string. The fallback to rptId is done later on view side -> https://github.com/pagopa/pagopa-checkout-fe/pull/364/files#diff-4ff917d9dac892053931fe565c0c5ba4f97054c9a5871ef6a1d507902207dd31R119. I think that creditorReferenceId is a different concept from RPT ID, so I'm saving both and show one on view side. (e.g. for example in the future I might want to show both). What do you think?